### PR TITLE
Init CodeQL analysis

### PR DESCRIPTION
**Pull request by jamiebuilds**
I'm not sure if this is worthwhile or not, but I was just going through repository settings and I figured I'd propose this.

For more info: https://github.com/github/codeql

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
```
Comment (ematipico) [inline .github/workflows/codeql-analysis.yml:22]: Should we build the project before? The entire project is in TypeScript. Does `javascript` cover TS too?

Comment (jamiebuilds) [reply .github/workflows/codeql-analysis.yml:22]: Yeah, they group javascript and typescript into one, it doesn't require building

**Comment by ematipico** [comment]
It would be great to see something, for some reason the operation timed out.

**Comment by jamiebuilds** [comment]
Yeah I'm not sure why its timing out. When you dig into the job it looks like it completed successfully

**Pull request closed without merging**